src/implementations/vanilla/components/modal/_modal.scss

```diff
@@ -0,0 +1,116 @@
+/**
+ * Modal
+ * @define modal
+ */
+
+@use 'sass:map';
+@use '@ecl/theme-dev/theme';
+@use '@ecl/vanilla-layout-grid/mixins/breakpoints';
+
+$_overlay: null !default;
+$_border-radius: null !default;
+$_separator-color: null !default;
+
+.ecl-modal {
+  background-color: $_overlay;
+  border: none;
+  color: map.get(theme.$color, 'grey-100');
+  height: 100%;
+  left: 0;
+  margin: 0;
+  max-height: 100%;
+  max-width: 100%;
+  padding: 0;
+  position: fixed;
+  top: 0;
+  width: 100%;
+}
+
+.ecl-modal[open] {
+  display: block;
+}
+
+.ecl-modal__container {
+  position: relative;
+  top: 50%;
+  transform: translateY(-50%);
+}
+
+.ecl-modal__content {
+  background-color: map.get(theme.$color, 'white-100');
+  border-radius: $_border-radius;
+  box-shadow: map.get(theme.$shadow, '4');
+  font: map.get(theme.$font-prolonged, 'm');
+  padding: 0;
+}
+
+.ecl-modal__header {
+  align-items: center;
```

Comment:
it feels it might work better with an "flex-start" value, in mobile:
![image](https://user-images.githubusercontent.com/14326885/221861952-1532236c-901d-4ae7-bb6a-e03226126d5d.png)